addons/account_edi/models/account_edi.py

```diff
@@ -0,0 +1,275 @@
+        """ Import an xml into an invoice.
+
+        :param tree:    The tree of the xml to import.
+        :param invoice: The invoice to set the decoded data.
+        :returns:       The created invoice.
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+        return self.env['account.move']
+
+    def _update_invoice_from_xml_tree(self, filename, tree, invoice):
+        """ Import an xml into an invoice.
+
+        :param tree:    The tree of the xml to import.
+        :param invoice: The invoice to set the decoded data.
+        :returns:       The updated invoice.
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+        return self.env['account.move']
+
+    def _create_invoice_from_pdf_reader(self, filename, reader):
+        """ Import an xml into an invoice.
+
+        :param reader:  The pdf reader of the pdf to import.
+        :param invoice: The invoice to set the decoded data.
+        :returns:       The created invoice.
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+
+        return self.env['account.move']
+
+    def _update_invoice_from_pdf_reader(self, filename, reader, invoice):
+        """ Import an xml into an invoice.
+
+        :param reader:  The pdf reader of the pdf to import.
+        :param invoice: The invoice to set the decoded data.
+        :returns:       The created invoice.
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+        return self.env['account.move']
+
+    ####################################################
+    # Export Internal methods (not meant to be overridden)
+    ####################################################
+
+    def _embed_attachments_into_pdf(self, pdf_content, invoice):
+        """ Create the EDI document of the invoice and embed it in the pdf_content.
+
+        :param pdf_content: the bytes representing the pdf to add the EDIs to.
+        :param invoice: the invoice to generate the EDI from.
+        :returns: the same pdf_content with the EDI of the invoice embed in it.
+        """
+        attachments = []
+        for edi_format in self.filtered(lambda e: e.export_invoice == 'embed_to_pdf'):
+            vals = edi_format._export_invoice_to_attachment(invoice)
+            if vals:
+                attachments.append(vals)
+
+        if attachments:
+            # Add the attachments to the pdf file
+            reader_buffer = io.BytesIO(pdf_content)
+            reader = OdooPdfFileReader(reader_buffer)
+            writer = OdooPdfFileWriter()
+            writer.cloneReaderDocumentRoot(reader)
+            for vals in attachments:
+                writer.addAttachment(vals['name'], vals['datas'])
+            buffer = io.BytesIO()
+            writer.write(buffer)
+            pdf_content = buffer.getvalue()
+            reader_buffer.close()
+            buffer.close()
+        return pdf_content
+
+    def _create_ir_attachments(self, invoice):
```

Comment:
We need a way to decide when the xml will be generated. This method is currently called when opening the "send by mail" wizard but some EDI will need to generate the xml when posting the invoice. What about having a method instead the field on account.edi.format like:
```
def _export_invoice_options(self):
  return {
    'export_type': 'embed_to_pdf' / 'separated_file',
    'export_generation': 'at_post' / 'at_send',
  }
```